weave/panels_py/generator_templates.py

```diff
@@ -0,0 +1,106 @@
+"""
+Contains the TemplateRegistry class, which is used to register
+templates for the PyBoard generator. I think we might want to make
+this even more generic and all any panel to be a generator, but for 
+now this is a simple abstraction that will work for basic use cases.
+
+# TODO: Generalize this to work with panels like /weave/ecosystem/wandb/panel_time_series.py
+# Proper panels have the following advantageous patterns:
+1. They can create default configs from inputs
+2. They can declare their own config editor rendering function
+3. Their render function is pre-defined
+4. If we make it work with the panel pattern, presumably it will be more general
+   and lend itself to ui-driven generation of panels.
+
+However, this current implementation is simple and easy to refactor.
+"""
+
+
+import dataclasses
+import typing
+
+from .. import graph
+from .. import weave_types
+from .. import registry_mem
+from .. import decorator_op
+
+
+@dataclasses.dataclass
+class TemplateRegistrySpec:
+    display_name: str
+    description: str
+    op_name: str
+    input_node_predicate: typing.Optional[
+        typing.Callable[[graph.Node[typing.Any]], bool]
+    ] = None
+    config_type: typing.Optional[weave_types.Type] = None
+
+
+@dataclasses.dataclass
+class _TemplateRegistry:
+    _specs: typing.Dict[str, TemplateRegistrySpec] = dataclasses.field(
+        default_factory=dict
+    )
+
+    def register(
+        self,
+        op_name: str,
+        display_name: str,
+        description: str,
+        input_node_predicate: typing.Optional[
+            typing.Callable[[graph.Node[typing.Any]], bool]
+        ] = None,
+        config_type: typing.Optional[weave_types.Type] = None,
+    ) -> None:
+        return self.register_spec(
+            TemplateRegistrySpec(
+                display_name=display_name,
+                description=description,
+                op_name=op_name,
+                input_node_predicate=input_node_predicate,
+                config_type=config_type,
+            )
+        )
+
+    def register_spec(self, spec: TemplateRegistrySpec) -> None:
+        if spec.op_name in self._specs:
+            raise ValueError(f"Template {spec.op_name} already registered")
+        self._specs[spec.op_name] = spec
+
+    def get_spec(self, name: str) -> typing.Any:
```

Comment:
Why is the result an `Any` and not a `TemplateRegistrySpec`?

Reply:
fixed